plugins/postgres/dbt/include/postgres/macros/adapters.sql

```diff
@@ -202,3 +202,13 @@
     comment on column {{ relation }}.{{ adapter.quote(column_name) if column_dict[column_name]['quote'] else column_name }} is {{ escaped_comment }};
   {% endfor %}
 {% endmacro %}
+
```

Comment:
Funny story: Postgres/Redshift use `grant {privilege} on table {relation}`, _regardless of whether `{relation}` is a table or a view_.

So Postgres will raise an error if we try to do `grant {privilege} on view {view_name}`. It must be `grant {privilege} on table {view_name}`.

Given that this is a wonkiness of PostgreSQL, I think the right move is to reimplement `postgres__get_grant_sql`. Notice that this is always `table`, instead of `{{ relation.type }}`:
```sql
{%- macro postgres__get_grant_sql(relation, grant_config) -%}
    {%- for privilege in grant_config.keys() -%}
        {%- set grantees = grant_config[privilege] -%}
        {%- if grantees -%}
            {%- for grantee in grantees %}
                grant {{ privilege }} on table {{ relation }} to {{ grantee}};
            {% endfor -%}
        {%- endif -%}
    {%- endfor -%}
{%- endmacro -%}
```